docs/rest_api.md

```diff
@@ -0,0 +1,558 @@
+
+
+---
+### `PATCH /annotations/:typ/:id/edit`
+
+Edit metadata of an annotation
+
+#### Expects
+ - In the url: `:typ` – one of `Task`, `Explorational`
+ - In the url: `:id` an annotation id
+ - JSON object with optional fields
+   - `"name"` `[STRING]` new name for the annotation
+   - `"description"` `[STRING]` new description for the annotation
+   - `"isPublic"` `[BOOLEAN]` whether or not the tracing should be shared publicly
+   - `"tags"` `[LIST[STRING]]` list of tags for the annotation
+
+#### Returns
+ - JSON object containing annotation information about the edited annotation
+
+
+---
+### `PATCH /annotations/:typ/:id/finish`
+
+#### Expects
+ - In the url: `:typ` – one of `Task`, `Explorational`
+ - In the url: `:id` an annotation id
+
+#### Returns
+ - JSON object containing annotation information about the finished annotation
+
+
+---
+### `PATCH /annotations/:typ/:id/reopen`
+
+#### Expects
+ - In the url: `:typ` – one of `Task`, `Explorational`
+ - In the url: `:id` an annotation id
+
+#### Returns
+ - JSON object containing annotation information about the reopened annotation
+
+
+---
+### `PUT /annotations/Task/:id/reset`
+
+Reset a task annotation to its base state
+
+#### Expects
+ - In the url: `:id` an annotation id
+
+#### Returns
+ - JSON object containing annotation information about the reset task annotation
+
+
+---
+### `POST /annotations/:typ/:id/merge/:mergedTyp/:mergedId`
+
+Merge two annotations, creating a new explorative.
+
+#### Expects
+ - In the url: `:typ` and `:id` as described above under `GET /annotations/:typ/:id/info`
+ - In the url: `:mergedTyp` and `:mergedId` as described above under `GET /annotations/:typ/:id/info`
+
+
+
+
+---
+### `POST /tasks`
+
+Create tasks without attached NML files
+
+#### Expects
+ - JSON list of objects. Each has these fields:
+   - `"taskTypeId"` `[STRING]`
+   - `"neededExperience"` `[JSON OBJECT]` with these fields:
+     - `"domain"` `[STRING]`
+     - `"value"` `[INT]`
+   - `"openInstances"` `[INT]`
+   - `"projectName"` `[STRING]`
+   - `"scriptId"` (optional) `[STRING]`
+   - `"boundingBox"` (optional) `[JSON OBJECT]` with these fields:
+     - `"topLeft"` `[JSON LIST OF THREE INTS]`
+     - `"width"` `[INT]`
+     - `"height"` `[INT]`
+     - `"depth"` `[INT]`
+   - `"dataSet"` `[STRING]` name of the dataset
+   - `"editPosition"` `[JSON LIST OF THREE INTS]` starting coordinates
+   - `"editRotation"` `[JSON LIST OF THREE INTS]` starting rotation
+   - `"creationInfo"` (optional) `[STRING]` identifier to match with task creation result report
+   - `"description"` (optional) `[STRING]`
+
+
+#### Returns
+
+ - JSON list with status reports about the created tasks, including their ids
+
+#### Note
+
+For each attached NML file, one task is created. This request will also return status code 200 if some tasks could not be created. Failed tasks are instead encoded in the JSON response
```

Comment:
I don't get this. Above it says "Create tasks without attached NML files" but here we talk about NMLs?